onnxruntime/core/mlas/inc/mlas.h

```diff
@@ -1128,3 +1128,21 @@ MlasQLinearMul(
     size_t N,
     bool IsScalarB
     );
+
+/**
+ * @brief Performs a single precision vector dot product operation.
+ *
+ * @param A          Supplies the A input vector of M length.
+ * @param TransB     Supplies the B input vector of M * N length.
+ * @param C          Supplies the C output vector of N length.
+ * @param M          Supplies the number of values in the A vector.
+ * @param N          Supplies the number of values in the C vector.
+ */
+void 
+MLASCALL
+MlasVectorDotProduct(
```

Comment:
It is 1xM and MxN matrix multiplication. It should serve the requirement:
[https://github.com/microsoft/onnxruntime/blob/1003db0058e1b08ebce89a25778816c75f610c47/onnxruntime/core/mlas/lib/sgemm.cpp#L1159](https://github.com/microsoft/onnxruntime/blob/1003db0058e1b08ebce89a25778816c75f610c47/onnxruntime/core/mlas/lib/sgemm.cpp#L1159)

Reply:
Thanks - good catch - closing this PR for now.